lib/livebook_web/live/output/input_component.ex

```diff
@@ -305,6 +440,49 @@ defmodule LivebookWeb.Output.InputComponent do
     {:ok, html_value}
   end
 
+  defp parse(html_value, %{type: :datetime}) do
+    case NaiveDateTime.from_iso8601(html_value) do
+      {:ok, datetime} -> {:ok, NaiveDateTime.truncate(datetime, :second)}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :time}) do
+    case Time.from_iso8601(html_value) do
+      {:ok, time} -> {:ok, Time.truncate(time, :second)}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :date}) do
+    case Date.from_iso8601(html_value) do
+      {:ok, date} -> {:ok, date}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :week}) do
+    try do
+      [year, week] = String.split(html_value, "-W")
+      {y, _} = Integer.parse(year)
+      {w, _} = Integer.parse(week)
+      {:ok, {y, w}}
```

Comment:
It hit me only now, but I'm not convinced what week/month is that useful. I don't think those are commonly used and it's easy to just have separate selects for week and year, etc. Date/time/datetime all make sense because we can parse into Elixir structs and those are common constructs.

Reply:
Yeah probably you are right, I removed them :+1: